launch/launch/launcher.py

```diff
@@ -51,13 +57,35 @@ def add_launch_descriptor(self, launch_descriptor):
 
             self.task_descriptors.append(task_descriptor)
 
+    def interrupt_launch(self):
+        with self.loop_lock:
+            if self.loop is not None:
+                self.loop.call_soon_threadsafe(self.interrupt_launch_non_threadsafe)
+
+    def interrupt_launch_non_threadsafe(self):
+        self.interrupt_future.set_result(True)
+
+    def wait_on_launch_to_finish(self, timeout):
+        return self.launch_complete.wait(timeout)
+
+    def launch_is_running(self):
+        return not self.launch_complete.is_set()
+
+    def wait_on_processes_to_launch(self, timeout):
+        return self.processes_launched.wait(timeout)
+
+    def processes_are_launched(self):
+        return self.processes_launched.is_set()
+
     def launch(self):
-        if os.name == 'nt':
-            # Windows needs a custom event loop to use subprocess transport
-            loop = asyncio.ProactorEventLoop()
-        else:
-            loop = asyncio.new_event_loop()
-        asyncio.set_event_loop(loop)
+        with self.loop_lock:
+            if os.name == 'nt':
+                # Windows needs a custom event loop to use subprocess transport
+                self.loop = asyncio.ProactorEventLoop()
+            else:
+                self.loop = asyncio.new_event_loop()
+            asyncio.set_event_loop(self.loop)
+        loop = self.loop
```

Comment:
Why does this block need a lock? I don't think calling `launch` multiple times should be a supported use case. With the `self.loop` variable being accessed outside the lock this method is not thread safe anyway.